### PR TITLE
raftstore-v2: fix delete range

### PR DESCRIPTION
**Pull request by tabokie**
### What is changed and how it works?

Issue Number: Ref #12842 
What's Changed:

```commit-message
- Do not use WAL for delete range writes
- Keep modified index consistent with memtable state, add tests for two possible anomalies.
```

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/raftstore-v2/src/operation/command/write/mod.rs [hunk 1]
```diff
@@ -295,19 +293,23 @@ impl<EK: KvEngine, R: ApplyResReporter> Apply<EK, R> {
...
             self.logger,
             "execute delete range";
             "range_start" => log_wrappers::Value::key(&start_key),
             "range_end" => log_wrappers::Value::key(&end_key),
             "notify_only" => notify_only,
-            "use_delete_range" => use_delete_range,
             "duration" => ?start.saturating_elapsed(),
         );
 
+        if index != u64::MAX && written {
```
Comment (overvenus) [inline components/raftstore-v2/src/operation/command/write/mod.rs:310]: When is it `u64::MAX`?

File: tests/integrations/raftstore/test_single.rs [hunk 2]
```diff
@@ -92,7 +92,7 @@ fn test_delete() {
 }
 
 #[test_case(test_raftstore::new_node_cluster)]
-#[test_case(test_raftstore_v2::new_node_cluster)]
+// v2 doesn't support.
```
Comment (overvenus) [inline tests/integrations/raftstore/test_single.rs:95]: Is it rocksdb delete range or RaftCmd::DeleteRange? Could you add some comments?

**Review by overvenus — COMMENTED** [review]
Rest LGTM

File: components/engine_rocks/src/misc.rs [hunk 3]
```diff
@@ -39,7 +42,8 @@ impl RocksEngine {
                 .as_ref()
                 .map_or(false, |key| key.as_slice() > r.start_key)
             {
-                self.delete_all_in_range_cf_by_key(cf, &r)?;
+                self.delete_all_in_range_cf_by_key(wopts, cf, &r)?;
+                written = true;
```
Comment (SpadeA-Tang) [inline components/engine_rocks/src/misc.rs:46]: written = self.delete_all_in_range_cf_by_key(wopts, cf, &r)?;

File: components/raftstore-v2/src/operation/command/write/mod.rs (hunk 1 above)
Comment (tabokie) [reply components/raftstore-v2/src/operation/command/write/mod.rs:310]: I don't know, it's the old code, shared by all write requests.

**Review by SpadeA-Tang — APPROVED** [review]

**Review by overvenus — APPROVED** [review]

**Pull request merged**